scorecard/profile_data.py

```diff
@@ -236,19 +241,79 @@ def revenue_breakdown(self):
                         'item': item,
                         'amount': amount,
                         'percent': percent(amount, total),
-                        'year': year_name,
+                        'date': year_name,
                     })
                 if total and subtotal and (total != subtotal):
                     values.append({
                         'item': 'Other',
                         'amount': total - subtotal,
                         'percent': percent(total - subtotal, total),
-                        'year': year_name,
+                        'date': year_name,
                     })
             except KeyError:
                 continue
         return {'values': values}
 
+    def current_ratio(self):
+        values = []
+        results = self.results['in_year_bsheet']
+        year_key = lambda r: r['financial_year_end.year']
+        month_key = lambda r: r['financial_period.period']
+        year_sorted = sorted(results, key=year_key)
+        year_sorted.reverse()
+        quarters = {}
+        latest_quarter = None
+        for year, yeargroup in groupby(year_sorted, year_key):
```

Comment:
I think you could do these two nested loops in one go by making the sort and group key compound: `(year, month)`. Python sorts tuples component-wise, which means `(2015, 4) < (2015, 3) < (2014, 4)`